### PR TITLE
improve docstring of Axes.plot_date

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

As part of #10148: Docstring update for `Axes.plot_date` and part of `matplotlib.dates`.

### CODE REVIEW

File: lib/matplotlib/dates.py [hunk 1]
```diff
@@ -20,24 +41,22 @@
    732403, whereas using the Gregorian calendar via the datetime
    module we find::
 
-     In [31]:date(2006,4,1).toordinal() - date(1,1,1).toordinal()
-     Out[31]:732401
+     In [1]: date(2006,4,1).toordinal() - date(1,1,1).toordinal()
```
Comment (anntzer) [inline lib/matplotlib/dates.py]: may as well make this pep8-compliant (with spaces after commas)?

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -1535,64 +1535,70 @@ def plot(self, *args, **kwargs):
...
-        to :meth:`plot_date` since :meth:`plot_date` will set the
-        default tick locator to
-        :class:`matplotlib.dates.AutoDateLocator` (if the tick
-        locator is not already set to a
-        :class:`matplotlib.dates.DateLocator` instance) and the
-        default tick formatter to
-        :class:`matplotlib.dates.AutoDateFormatter` (if the tick
-        formatter is not already set to a
-        :class:`matplotlib.dates.DateFormatter` instance).
+        Note: If you are using custom date tickers and formatters, it may be
```
Comment (dstansby) [inline lib/matplotlib/axes/_axes.py]: Can this go in the notes section below?

**Comment by jklymak** [comment]
Push committed the minor change, though @timhoffm let me know if that ruins your workflow, and I won't do again.  Sorry for not asking first - I saw "dstansby" and my mind got confused.

**Comment by tacaswell** [comment]
backported to v2.1.x as as 6d9eab7baaf39192a748ef27eee26469ab646b76

@timhoffm Sorry, I missed that you had opened a PR to do this as well!

**Pull request merged**